// @buf/googleapis_googleapis.community_timostamm-protobuf-ts/google/api/client_pb.js
/* eslint-disable */
// @generated by protobuf-ts 2.8.2 with parameter output_javascript,optimize_code_size,long_type_string,add_pb_suffix,ts_nocheck,eslint_disable
// @generated from protobuf file "google/api/client.proto" (package "google.api", syntax proto3)
// tslint:disable
// @ts-nocheck
//
// Copyright 2018 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
/* eslint-disable */
// @generated by protobuf-ts 2.8.2 with parameter output_javascript,optimize_code_size,long_type_string,add_pb_suffix,ts_nocheck,eslint_disable
// @generated from protobuf file "google/api/client.proto" (package "google.api", syntax proto3)
// tslint:disable
// @ts-nocheck
//
// Copyright 2018 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
import { MessageType } from "@protobuf-ts/runtime";
import { Duration } from "../protobuf/duration_pb";
import { LaunchStage } from "./launch_stage_pb";
/**
 * The organization for which the client libraries are being published.
 * Affects the url where generated docs are published, etc.
 *
 * @generated from protobuf enum google.api.ClientLibraryOrganization
 */
export var ClientLibraryOrganization;
(function (ClientLibraryOrganization) {
    /**
     * Not useful.
     *
     * @generated from protobuf enum value: CLIENT_LIBRARY_ORGANIZATION_UNSPECIFIED = 0;
     */
    ClientLibraryOrganization[ClientLibraryOrganization["CLIENT_LIBRARY_ORGANIZATION_UNSPECIFIED"] = 0] = "CLIENT_LIBRARY_ORGANIZATION_UNSPECIFIED";
    /**
     * Google Cloud Platform Org.
     *
     * @generated from protobuf enum value: CLOUD = 1;
     */
    ClientLibraryOrganization[ClientLibraryOrganization["CLOUD"] = 1] = "CLOUD";
    /**
     * Ads (Advertising) Org.
     *
     * @generated from protobuf enum value: ADS = 2;
     */
    ClientLibraryOrganization[ClientLibraryOrganization["ADS"] = 2] = "ADS";
    /**
     * Photos Org.
     *
     * @generated from protobuf enum value: PHOTOS = 3;
     */
    ClientLibraryOrganization[ClientLibraryOrganization["PHOTOS"] = 3] = "PHOTOS";
    /**
     * Street View Org.
     *
     * @generated from protobuf enum value: STREET_VIEW = 4;
     */
    ClientLibraryOrganization[ClientLibraryOrganization["STREET_VIEW"] = 4] = "STREET_VIEW";
})(ClientLibraryOrganization || (ClientLibraryOrganization = {}));
/**
 * To where should client libraries be published?
 *
 * @generated from protobuf enum google.api.ClientLibraryDestination
 */
export var ClientLibraryDestination;
(function (ClientLibraryDestination) {
    /**
     * Client libraries will neither be generated nor published to package
     * managers.
     *
     * @generated from protobuf enum value: CLIENT_LIBRARY_DESTINATION_UNSPECIFIED = 0;
     */
    ClientLibraryDestination[ClientLibraryDestination["CLIENT_LIBRARY_DESTINATION_UNSPECIFIED"] = 0] = "CLIENT_LIBRARY_DESTINATION_UNSPECIFIED";
    /**
     * Generate the client library in a repo under github.com/googleapis,
     * but don't publish it to package managers.
     *
     * @generated from protobuf enum value: GITHUB = 10;
     */
    ClientLibraryDestination[ClientLibraryDestination["GITHUB"] = 10] = "GITHUB";
    /**
     * Publish the library to package managers like nuget.org and npmjs.com.
     *
     * @generated from protobuf enum value: PACKAGE_MANAGER = 20;
     */
    ClientLibraryDestination[ClientLibraryDestination["PACKAGE_MANAGER"] = 20] = "PACKAGE_MANAGER";
})(ClientLibraryDestination || (ClientLibraryDestination = {}));
// @generated message type with reflection information, may provide speed optimized methods
class CommonLanguageSettings$Type extends MessageType {
    constructor() {
        super("google.api.CommonLanguageSettings", [
            { no: 1, name: "reference_docs_uri", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "destinations", kind: "enum", repeat: 1 /*RepeatType.PACKED*/, T: () => ["google.api.ClientLibraryDestination", ClientLibraryDestination] }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message google.api.CommonLanguageSettings
 */
export const CommonLanguageSettings = new CommonLanguageSettings$Type();
// @generated message type with reflection information, may provide speed optimized methods
class ClientLibrarySettings$Type extends MessageType {
    constructor() {
        super("google.api.ClientLibrarySettings", [
            { no: 1, name: "version", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "launch_stage", kind: "enum", T: () => ["google.api.LaunchStage", LaunchStage] },
            { no: 3, name: "rest_numeric_enums", kind: "scalar", T: 8 /*ScalarType.BOOL*/ },
            { no: 21, name: "java_settings", kind: "message", T: () => JavaSettings },
            { no: 22, name: "cpp_settings", kind: "message", T: () => CppSettings },
            { no: 23, name: "php_settings", kind: "message", T: () => PhpSettings },
            { no: 24, name: "python_settings", kind: "message", T: () => PythonSettings },
            { no: 25, name: "node_settings", kind: "message", T: () => NodeSettings },
            { no: 26, name: "dotnet_settings", kind: "message", T: () => DotnetSettings },
            { no: 27, name: "ruby_settings", kind: "message", T: () => RubySettings },
            { no: 28, name: "go_settings", kind: "message", T: () => GoSettings }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message google.api.ClientLibrarySettings
 */
export const ClientLibrarySettings = new ClientLibrarySettings$Type();
// @generated message type with reflection information, may provide speed optimized methods
class Publishing$Type extends MessageType {
    constructor() {
        super("google.api.Publishing", [
            { no: 2, name: "method_settings", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => MethodSettings },
            { no: 101, name: "new_issue_uri", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 102, name: "documentation_uri", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 103, name: "api_short_name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 104, name: "github_label", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 105, name: "codeowner_github_teams", kind: "scalar", repeat: 2 /*RepeatType.UNPACKED*/, T: 9 /*ScalarType.STRING*/ },
            { no: 106, name: "doc_tag_prefix", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 107, name: "organization", kind: "enum", T: () => ["google.api.ClientLibraryOrganization", ClientLibraryOrganization] },
            { no: 109, name: "library_settings", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => ClientLibrarySettings }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message google.api.Publishing
 */
export const Publishing = new Publishing$Type();
// @generated message type with reflection information, may provide speed optimized methods
class JavaSettings$Type extends MessageType {
    constructor() {
        super("google.api.JavaSettings", [
            { no: 1, name: "library_package", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "service_class_names", kind: "map", K: 9 /*ScalarType.STRING*/, V: { kind: "scalar", T: 9 /*ScalarType.STRING*/ } },
            { no: 3, name: "common", kind: "message", T: () => CommonLanguageSettings }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message google.api.JavaSettings
 */
export const JavaSettings = new JavaSettings$Type();
// @generated message type with reflection information, may provide speed optimized methods
class CppSettings$Type extends MessageType {
    constructor() {
        super("google.api.CppSettings", [
            { no: 1, name: "common", kind: "message", T: () => CommonLanguageSettings }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message google.api.CppSettings
 */
export const CppSettings = new CppSettings$Type();
// @generated message type with reflection information, may provide speed optimized methods
class PhpSettings$Type extends MessageType {
    constructor() {
        super("google.api.PhpSettings", [
            { no: 1, name: "common", kind: "message", T: () => CommonLanguageSettings }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message google.api.PhpSettings
 */
export const PhpSettings = new PhpSettings$Type();
// @generated message type with reflection information, may provide speed optimized methods
class PythonSettings$Type extends MessageType {
    constructor() {
        super("google.api.PythonSettings", [
            { no: 1, name: "common", kind: "message", T: () => CommonLanguageSettings }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message google.api.PythonSettings
 */
export const PythonSettings = new PythonSettings$Type();
// @generated message type with reflection information, may provide speed optimized methods
class NodeSettings$Type extends MessageType {
    constructor() {
        super("google.api.NodeSettings", [
            { no: 1, name: "common", kind: "message", T: () => CommonLanguageSettings }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message google.api.NodeSettings
 */
export const NodeSettings = new NodeSettings$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DotnetSettings$Type extends MessageType {
    constructor() {
        super("google.api.DotnetSettings", [
            { no: 1, name: "common", kind: "message", T: () => CommonLanguageSettings }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message google.api.DotnetSettings
 */
export const DotnetSettings = new DotnetSettings$Type();
// @generated message type with reflection information, may provide speed optimized methods
class RubySettings$Type extends MessageType {
    constructor() {
        super("google.api.RubySettings", [
            { no: 1, name: "common", kind: "message", T: () => CommonLanguageSettings }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message google.api.RubySettings
 */
export const RubySettings = new RubySettings$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GoSettings$Type extends MessageType {
    constructor() {
        super("google.api.GoSettings", [
            { no: 1, name: "common", kind: "message", T: () => CommonLanguageSettings }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message google.api.GoSettings
 */
export const GoSettings = new GoSettings$Type();
// @generated message type with reflection information, may provide speed optimized methods
class MethodSettings$Type extends MessageType {
    constructor() {
        super("google.api.MethodSettings", [
            { no: 1, name: "selector", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "long_running", kind: "message", T: () => MethodSettings_LongRunning }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message google.api.MethodSettings
 */
export const MethodSettings = new MethodSettings$Type();
// @generated message type with reflection information, may provide speed optimized methods
class MethodSettings_LongRunning$Type extends MessageType {
    constructor() {
        super("google.api.MethodSettings.LongRunning", [
            { no: 1, name: "initial_poll_delay", kind: "message", T: () => Duration },
            { no: 2, name: "poll_delay_multiplier", kind: "scalar", T: 2 /*ScalarType.FLOAT*/ },
            { no: 3, name: "max_poll_delay", kind: "message", T: () => Duration },
            { no: 4, name: "total_poll_timeout", kind: "message", T: () => Duration }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message google.api.MethodSettings.LongRunning
 */
export const MethodSettings_LongRunning = new MethodSettings_LongRunning$Type();
