// @buf/darvand_fintrackapis.community_timostamm-protobuf-ts/transaction/v1/transaction_pb.js
/* eslint-disable */
// @generated by protobuf-ts 2.8.2 with parameter output_javascript,optimize_code_size,long_type_string,add_pb_suffix,ts_nocheck,eslint_disable
// @generated from protobuf file "transaction/v1/transaction.proto" (package "transaction.v1", syntax proto3)
// tslint:disable
// @ts-nocheck
/* eslint-disable */
// @generated by protobuf-ts 2.8.2 with parameter output_javascript,optimize_code_size,long_type_string,add_pb_suffix,ts_nocheck,eslint_disable
// @generated from protobuf file "transaction/v1/transaction.proto" (package "transaction.v1", syntax proto3)
// tslint:disable
// @ts-nocheck
import { MessageType } from "@protobuf-ts/runtime";
import { DateTime } from "@buf/googleapis_googleapis.community_timostamm-protobuf-ts/google/type/datetime_pb";
/**
 * @generated from protobuf enum transaction.v1.PaymentMethod
 */
export var PaymentMethod;
(function (PaymentMethod) {
    /**
     * @generated from protobuf enum value: PAYMENT_METHOD_UNSPECIFIED = 0;
     */
    PaymentMethod[PaymentMethod["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from protobuf enum value: PAYMENT_METHOD_CREDIT = 1;
     */
    PaymentMethod[PaymentMethod["CREDIT"] = 1] = "CREDIT";
    /**
     * @generated from protobuf enum value: PAYMENT_METHOD_DEBIT = 2;
     */
    PaymentMethod[PaymentMethod["DEBIT"] = 2] = "DEBIT";
    /**
     * @generated from protobuf enum value: PAYMENT_METHOD_CASH = 3;
     */
    PaymentMethod[PaymentMethod["CASH"] = 3] = "CASH";
})(PaymentMethod || (PaymentMethod = {}));
/**
 * @generated from protobuf enum transaction.v1.Category
 */
export var Category;
(function (Category) {
    /**
     * @generated from protobuf enum value: CATEGORY_UNSPECIFIED = 0;
     */
    Category[Category["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from protobuf enum value: CATEGORY_GROCERIES = 1;
     */
    Category[Category["GROCERIES"] = 1] = "GROCERIES";
    /**
     * @generated from protobuf enum value: CATEGORY_FOOD = 2;
     */
    Category[Category["FOOD"] = 2] = "FOOD";
    /**
     * @generated from protobuf enum value: CATEGORY_CLOTHS = 3;
     */
    Category[Category["CLOTHS"] = 3] = "CLOTHS";
    /**
     * @generated from protobuf enum value: CATEGORY_PETS = 4;
     */
    Category[Category["PETS"] = 4] = "PETS";
    /**
     * @generated from protobuf enum value: CATEGORY_HOUSE = 5;
     */
    Category[Category["HOUSE"] = 5] = "HOUSE";
})(Category || (Category = {}));
// @generated message type with reflection information, may provide speed optimized methods
class Transaction$Type extends MessageType {
    constructor() {
        super("transaction.v1.Transaction", [
            { no: 1, name: "id", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "source", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "value", kind: "scalar", T: 13 /*ScalarType.UINT32*/ },
            { no: 4, name: "date", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 5, name: "payment_method", kind: "enum", T: () => ["transaction.v1.PaymentMethod", PaymentMethod, "PAYMENT_METHOD_"] },
            { no: 6, name: "account_id", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 7, name: "user_id", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 8, name: "details", kind: "message", repeat: 1 /*RepeatType.PACKED*/, T: () => TransactionDetail },
            { no: 9, name: "created_at", kind: "message", T: () => DateTime },
            { no: 10, name: "updated_at", kind: "message", T: () => DateTime }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message transaction.v1.Transaction
 */
export const Transaction = new Transaction$Type();
// @generated message type with reflection information, may provide speed optimized methods
class TransactionDetail$Type extends MessageType {
    constructor() {
        super("transaction.v1.TransactionDetail", [
            { no: 1, name: "id", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 2, name: "quantity", kind: "scalar", T: 13 /*ScalarType.UINT32*/ },
            { no: 3, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "value", kind: "scalar", T: 5 /*ScalarType.INT32*/ },
            { no: 5, name: "category", kind: "enum", T: () => ["transaction.v1.Category", Category, "CATEGORY_"] },
            { no: 6, name: "total", kind: "scalar", T: 13 /*ScalarType.UINT32*/ },
            { no: 7, name: "unit_value", kind: "scalar", T: 13 /*ScalarType.UINT32*/ },
            { no: 8, name: "brand", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message transaction.v1.TransactionDetail
 */
export const TransactionDetail = new TransactionDetail$Type();
