// @buf/darvand_petapis.community_timostamm-protobuf-ts/pet/v1/pet_pb.js
/* eslint-disable */
// @generated by protobuf-ts 2.8.2 with parameter output_javascript,optimize_code_size,long_type_string,add_pb_suffix,ts_nocheck,eslint_disable
// @generated from protobuf file "pet/v1/pet.proto" (package "pet.v1", syntax proto3)
// tslint:disable
// @ts-nocheck
/* eslint-disable */
// @generated by protobuf-ts 2.8.2 with parameter output_javascript,optimize_code_size,long_type_string,add_pb_suffix,ts_nocheck,eslint_disable
// @generated from protobuf file "pet/v1/pet.proto" (package "pet.v1", syntax proto3)
// tslint:disable
// @ts-nocheck
import { ServiceType } from "@protobuf-ts/runtime-rpc";
import { MessageType } from "@protobuf-ts/runtime";
import { DateTime } from "@buf/googleapis_googleapis.community_timostamm-protobuf-ts/google/type/datetime_pb";
/**
 * PetType represents the different types of pets in the pet store.
 *
 * @generated from protobuf enum pet.v1.PetType
 */
export var PetType;
(function (PetType) {
    /**
     * @generated from protobuf enum value: PET_TYPE_UNSPECIFIED = 0;
     */
    PetType[PetType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /**
     * @generated from protobuf enum value: PET_TYPE_CAT = 1;
     */
    PetType[PetType["CAT"] = 1] = "CAT";
    /**
     * @generated from protobuf enum value: PET_TYPE_DOG = 2;
     */
    PetType[PetType["DOG"] = 2] = "DOG";
    /**
     * @generated from protobuf enum value: PET_TYPE_SNAKE = 3;
     */
    PetType[PetType["SNAKE"] = 3] = "SNAKE";
    /**
     * @generated from protobuf enum value: PET_TYPE_HAMSTER = 4;
     */
    PetType[PetType["HAMSTER"] = 4] = "HAMSTER";
})(PetType || (PetType = {}));
// @generated message type with reflection information, may provide speed optimized methods
class Pet$Type extends MessageType {
    constructor() {
        super("pet.v1.Pet", [
            { no: 1, name: "pet_type", kind: "enum", T: () => ["pet.v1.PetType", PetType, "PET_TYPE_"] },
            { no: 2, name: "pet_id", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 3, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ },
            { no: 4, name: "created_at", kind: "message", T: () => DateTime }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message pet.v1.Pet
 */
export const Pet = new Pet$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GetPetRequest$Type extends MessageType {
    constructor() {
        super("pet.v1.GetPetRequest", [
            { no: 1, name: "pet_id", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message pet.v1.GetPetRequest
 */
export const GetPetRequest = new GetPetRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class GetPetResponse$Type extends MessageType {
    constructor() {
        super("pet.v1.GetPetResponse", [
            { no: 1, name: "pet", kind: "message", T: () => Pet }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message pet.v1.GetPetResponse
 */
export const GetPetResponse = new GetPetResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class PutPetRequest$Type extends MessageType {
    constructor() {
        super("pet.v1.PutPetRequest", [
            { no: 1, name: "pet_type", kind: "enum", T: () => ["pet.v1.PetType", PetType, "PET_TYPE_"] },
            { no: 2, name: "name", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message pet.v1.PutPetRequest
 */
export const PutPetRequest = new PutPetRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class PutPetResponse$Type extends MessageType {
    constructor() {
        super("pet.v1.PutPetResponse", [
            { no: 1, name: "pet", kind: "message", T: () => Pet }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message pet.v1.PutPetResponse
 */
export const PutPetResponse = new PutPetResponse$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DeletePetRequest$Type extends MessageType {
    constructor() {
        super("pet.v1.DeletePetRequest", [
            { no: 1, name: "pet_id", kind: "scalar", T: 9 /*ScalarType.STRING*/ }
        ]);
    }
}
/**
 * @generated MessageType for protobuf message pet.v1.DeletePetRequest
 */
export const DeletePetRequest = new DeletePetRequest$Type();
// @generated message type with reflection information, may provide speed optimized methods
class DeletePetResponse$Type extends MessageType {
    constructor() {
        super("pet.v1.DeletePetResponse", []);
    }
}
/**
 * @generated MessageType for protobuf message pet.v1.DeletePetResponse
 */
export const DeletePetResponse = new DeletePetResponse$Type();
/**
 * @generated ServiceType for protobuf service pet.v1.PetStoreService
 */
export const PetStoreService = new ServiceType("pet.v1.PetStoreService", [
    { name: "GetPet", options: {}, I: GetPetRequest, O: GetPetResponse },
    { name: "PutPet", options: {}, I: PutPetRequest, O: PutPetResponse },
    { name: "DeletePet", options: {}, I: DeletePetRequest, O: DeletePetResponse }
]);
