// @buf/darvand_petapis.community_timostamm-protobuf-ts/pet/v1/pet_pb.client.js
/* eslint-disable */
// @generated by protobuf-ts 2.8.2 with parameter output_javascript,optimize_code_size,long_type_string,add_pb_suffix,ts_nocheck,eslint_disable
// @generated from protobuf file "pet/v1/pet.proto" (package "pet.v1", syntax proto3)
// tslint:disable
// @ts-nocheck
import { PetStoreService } from "./pet_pb";
import { stackIntercept } from "@protobuf-ts/runtime-rpc";
/**
 * @generated from protobuf service pet.v1.PetStoreService
 */
export class PetStoreServiceClient {
    constructor(_transport) {
        this._transport = _transport;
        this.typeName = PetStoreService.typeName;
        this.methods = PetStoreService.methods;
        this.options = PetStoreService.options;
    }
    /**
     * @generated from protobuf rpc: GetPet(pet.v1.GetPetRequest) returns (pet.v1.GetPetResponse);
     */
    getPet(input, options) {
        const method = this.methods[0], opt = this._transport.mergeOptions(options);
        return stackIntercept("unary", this._transport, method, opt, input);
    }
    /**
     * @generated from protobuf rpc: PutPet(pet.v1.PutPetRequest) returns (pet.v1.PutPetResponse);
     */
    putPet(input, options) {
        const method = this.methods[1], opt = this._transport.mergeOptions(options);
        return stackIntercept("unary", this._transport, method, opt, input);
    }
    /**
     * @generated from protobuf rpc: DeletePet(pet.v1.DeletePetRequest) returns (pet.v1.DeletePetResponse);
     */
    deletePet(input, options) {
        const method = this.methods[2], opt = this._transport.mergeOptions(options);
        return stackIntercept("unary", this._transport, method, opt, input);
    }
}
